### ISSUE TITLE
Add 'Download All' option for multi-file Works

### ISSUE THREAD

**Issue by cpd3149** [samvera/hyrax#1203]
### Descriptive summary

Relates to Hyrax 1.x

### Rationale

Users may want to download all of the files in a Work with one click, such as [this example of MATLAB scripts and formatted data](https://arch.library.northwestern.edu/concern/generic_works/bv73c042m). 

Story: Reader reads an article that lists the DOI of the article's supporting data set. Reader visits the data set in the institutional repository, represented as a Work. The Work contains X number of files. Reader wants to download a local copy of the data set.

### Expected behavior

Here's an example from [Figshare](https://figshare.com/articles/Supporting_Fileset_The_type_localities_of_Anolis_aequatorialis_Werner_1894_Sauria_Iguania_Dactyloidae_and_Pristimantis_appendiculatus_Werner_1894_Amphibia_Anura_Craugastoridae_/3753387): 

![image](https://user-images.githubusercontent.com/24395592/27053607-75063dc2-4f83-11e7-86ee-86522519c25d.png)

Here's an example from a [journal](http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0177347#sec036):

![image](https://user-images.githubusercontent.com/24395592/27053655-a3edd56e-4f83-11e7-9bdf-9ee6a981898f.png)

### Actual behavior

Download each file individually.

### Related work

If your local instance supports this functionality, please let us know.

**Comment by cpd3149** [comment]
There's the question of nested works. I'm not sure I understand nested works (chapters within a book?) and how they might relate to files within a work. My hunch is to limit this feature to downloading all the files within a single works.

**Comment by escowles** [comment]
I agree that this feature would be useful even if it was limited to the files in a single work.

I think some nested works might be better if this feature was recursive.  For example, we have MultiVolumeWorks, which bundle multiple ScannedResources, where each ScannedResource is a volume in a multi-volume book set.  Being able to download the entire set in a single zip file would be better, but being able to download each volume as a zip file would be better than nothing.

**Comment by mjgiarlo** [comment]
Your thoughts about technical design are welcome, @samvera/hyrax-code-reviewers 

And your thoughts about how to make this a pleasant UX are also welcome, @samvera/hyrax-ui-ux-advisors

**Comment by blancoj** [comment]
We added this to our repository.  You can take a look at the way it works
here:

https://deepblue.lib.umich.edu/data/concern/generic_works/3f462547h

There was an initial PR for this and then two others to fix some
bugs/changes:

https://github.com/mlibrary/umrdr/pulls?utf8=%E2%9C%93&q=is%3Apr%20is%3Aclosed%20download%20all

**Comment by newmanld** [comment]
Good functionality to add.   "Download All Files" as shown on DeepBlue.lib.umich.edu is simple and intuitive.  

I agree that even if it does not extend to child works that it is a feature worth adding.

Is it out of scope to provide the metadata of the work as a csv (or xml) included in the zip?  That metadata may contain important rights information as well as identify the works.  The feature would also then apply to works with a single file attached.

**Comment by hannahfrost** [comment]
:+1: to download all files + metadata!

**Comment by julesies** [comment]
Re-upping as important to repo-managers!
